programs/stake/src/stake_state.rs

```diff
@@ -560,296 +155,203 @@ pub struct PointValue {
-        .map(|(stakers_reward, voters_reward, credits_observed)| {
-            if let Some(inflation_point_calc_tracer) = inflation_point_calc_tracer {
-                inflation_point_calc_tracer(&InflationPointCalculationEvent::CreditsObserved(
-                    self.credits_observed,
-                    Some(credits_observed),
-                ));
-            }
-            self.credits_observed = credits_observed;
-            self.delegation.stake += stakers_reward;
-            (stakers_reward, voters_reward)
-        })
-    }
+        stake.credits_observed = credits_observed;
+        stake.delegation.stake += stakers_reward;
+        (stakers_reward, voters_reward)
+    })
+}
 
-    pub fn calculate_points(
-        &self,
-        vote_state: &VoteState,
-        stake_history: Option<&StakeHistory>,
-        inflation_point_calc_tracer: &mut Option<impl FnMut(&InflationPointCalculationEvent)>,
-        fix_stake_deactivate: bool,
-    ) -> u128 {
-        self.calculate_points_and_credits(
-            vote_state,
-            stake_history,
-            inflation_point_calc_tracer,
-            fix_stake_deactivate,
-        )
-        .0
-    }
+fn calculate_stake_points(
+    stake: &Stake,
+    vote_state: &VoteState,
+    stake_history: Option<&StakeHistory>,
+    inflation_point_calc_tracer: &mut Option<impl FnMut(&InflationPointCalculationEvent)>,
+    fix_stake_deactivate: bool,
+) -> u128 {
+    calculate_points_and_credits(
+        stake,
+        vote_state,
+        stake_history,
+        inflation_point_calc_tracer,
+        fix_stake_deactivate,
+    )
+    .0
+}
 
-    /// for a given stake and vote_state, calculate how many
-    ///   points were earned (credits * stake) and new value
-    ///   for credits_observed were the points paid
-    fn calculate_points_and_credits(
-        &self,
-        new_vote_state: &VoteState,
-        stake_history: Option<&StakeHistory>,
-        inflation_point_calc_tracer: &mut Option<impl FnMut(&InflationPointCalculationEvent)>,
-        fix_stake_deactivate: bool,
-    ) -> (u128, u64) {
-        // if there is no newer credits since observed, return no point
-        if new_vote_state.credits() <= self.credits_observed {
-            if fix_stake_deactivate {
-                if let Some(inflation_point_calc_tracer) = inflation_point_calc_tracer {
-                    inflation_point_calc_tracer(&SkippedReason::ZeroCreditsAndReturnCurrent.into());
-                }
-                return (0, self.credits_observed);
-            } else {
-                if let Some(inflation_point_calc_tracer) = inflation_point_calc_tracer {
-                    inflation_point_calc_tracer(&SkippedReason::ZeroCreditsAndReturnZero.into());
-                }
-                return (0, 0);
+/// for a given stake and vote_state, calculate how many
+///   points were earned (credits * stake) and new value
+///   for credits_observed were the points paid
+fn calculate_points_and_credits(
```

Comment:
```suggestion
fn calculate_stake_points_and_credits(
```
to be consistent with the other function name changes?

Reply:
Nice catch, yep will do